### PR TITLE
	Make message allocation infallible, and reap some benefits

### LINKED ISSUE #118: Don't return Result from Message construction
According to the manpages, the only error that can occur when creating a new Message is out-of-memory, but Rust policy is to panic in these cases.

This is another API incompability, therefore for 0.9?

### PR DESCRIPTION
**Pull request by rotty**
This fixes #118, and allows for implementation of the `From<&[u8]>`
and `From<Vec<u8>` for `Message`.

### CODE REVIEW

**Comment by birkenfeld** [comment]
Looks good to me.

File: src/lib.rs [hunk 1]
```diff
@@ -597,8 +597,7 @@ impl Socket {
 
     /// Send a `&[u8]` message.
     pub fn send(&self, data: &[u8], flags: i32) -> Result<()> {
```
Comment (birkenfeld) [inline src/lib.rs]: Could/should this take `Into<Message>`? Then `send_msg` could be retired.

Comment (rotty) [inline src/lib.rs]: Very good point. I'll change that.

Comment (rotty) [inline src/lib.rs]: I've now added that, including more `From` implementation for `Message`, and simplifying call sites accordingly.

**Pull request merged**